docs/design/Path concatenation.md

```diff
@@ -0,0 +1,59 @@
+## Paths in DF2
+
+It is expected that the use of paths in DF2 (similar in nature to Unix file paths, but not exactly the same) will be extensive, as will be concatention operations on them: `GetFullPath` and `GetRelativePath`. This document will provide a specification for how these operations are to behave as well as contain some considerations that will drive the design of concatenation functions.
+
+## Definitions
+
+Fundamentally, DF2 _paths_ are _strings_ that consist of _segments_ that are separated by _separators_. Two _paths_ are equal if all their _segments_ are equal. _Paths_ can have both _trailing_ and _leading_ _separators_.
+
+_Strings_ are arrays of `char`s that are interpreted as UTF-16 encoded character sequences and have a trailing zero `char`. _Strings_ do not need to be valid UTF-16.
```

Comment:
In binary they're stored in UTF-8 and are read as a null-terminated UTF-8 string. It's only .NET's representation of those strings that are UTF-16.

Reply:
Hmm, would it be fair to add that the spec is about .NET in-process (UTF16) only representation? Or is the plan to eventually migrate to using the equivalent of `byte[]` in-process as well?

Reply:
To be fair we could use byte arrays, but for the purposes of the spec I think I'd just leave the definition of "string" as implementation-defined so long as it can store the UTF-8 characters retrieved from the data.

Reply:
That's fair. I will reword the spec in terms of code point sequences.